packages/documentation/docs/for-developers/for-blueprint-developers/ab-playback.md

```diff
@@ -1,100 +1,103 @@
 # AB Playback
 
-:::caution
-This page talks about various sections that are in NRK's blueprints which are not yet public. Other blueprints (such as those from [TV 2 Danmark](https://github.com/tv2/sofie-blueprints-inews)) make use of AB playback, but may not match up completely with what is documented here.  
-We are looking into pushing more of this logic into Sofie to aid reusability.
+:::info
+Prior to 1.50 of Sofie, this was implemented in Blueprints and not natively in Sofie-core
 :::
 
 _AB Playback_ is a common technique for clip playback. The aim is to be able to play multiple clips back to back, alternating which player is used for each clip.  
 At first glance it sounds simple to handle, but it quickly becomes complicated when we consider the need to allow users to run adlibs and that the system needs to seamlessly update pre-programmed clips when this happens.
 
-To avoid this problem, we take an approach of simply labelling pieces as needing an AB assignment and leaving timeline objects to have some unresolved values during the ingest blueprint operations, and use the `onTimelineGenerate` blueprint function to do the assignment and modify the timeline as needed at the time of playout.
+To avoid this problem, we take an approach of labelling pieces as needing an AB assignment and leaving timeline objects to have some unresolved values during the ingest blueprint operations, and we perform the AB resolving when building the timeline for playout.
 
-There are other challenges to think about too:
+There are other challenges to the resolving to think about too, which make this a challenging area to tackle, and not something that wants to be considered when starting out with blueprints. Some of these challenges are:
 
 - Users get confused if the player of a clip changes without a reason
 - Reloading an already loaded clip can be costly, so should be avoided when possible
 - Adlibbing a clip, or changing what Part is nexted can result in needing to move what player a clip has assigned
 - Postroll or preroll is often needed
 - Some studios can have less players available than ideal. (eg, going back to back between two clips, and a clip is playing on the studio monitor)
 
-## Defining
+## Defining Piece sessions
 
-:::info
-This method of defining sessions is just a suggestion, it could be done in many other ways depending on how you wish to implement it.
-:::
+In order for the AB resolver to know what AB sessions there are on the timeline, and how they all relate to each other, we define `abSessions` properties on various objects when defining Pieces and their content during the `getSegment` blueprint method.
```

Comment:
I think it would be nice to have a sentence or two about what a _Session_ is, in this context.